File: 2.0/WearOS/WatchSprings-2.0/app/src/main/java/copyright/markmorris/allrightsreserved/watchsprings/screens/NewsDetailScreen.kt
// WatchSprings 2.0 © 2025 Mark S. Morris. All rights reserved.
//
// https://www.watchsprings.com
//
// WatchSprings is Fan Content pursuant to the Epic Games Fan Content Policy. Portions of the materials used are trademarks and/or copyrighted works of Epic Games, Inc.

package copyright.markmorris.allrightsreserved.watchsprings

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.wear.compose.foundation.lazy.ScalingLazyColumn
import androidx.wear.compose.material.MaterialTheme
import androidx.wear.compose.material.Text
import coil.compose.SubcomposeAsyncImage

@Composable
fun NewsDetailScreen(viewModel: NewsViewModel)
{
    val news by viewModel.selectedNews.collectAsState()

    ChildScreen(navController = null, title = news?.title ?: "News Detail") { scalingLazyListState ->
        ScalingLazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(8.dp),
            state = scalingLazyListState
        ) {
            if (news != null) {
                item {
                    SubcomposeAsyncImage(
                        model = news!!.image,
                        contentDescription = null,
                        contentScale = ContentScale.FillWidth,
                        loading = {
                            Box(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .aspectRatio(16f / 9f)
                                    .background(Color.DarkGray)
                            )
                        },
                        modifier = Modifier
                            .fillMaxWidth()
                            .aspectRatio(16f / 9f)
                    )
                }

                item {
                    Text(
                        text = news!!.body,
                        style = MaterialTheme.typography.body1,
                        textAlign = TextAlign.Center,
                        modifier = Modifier.fillMaxWidth(),
                        color = Color.White
                    )
                }
            } else {
                item {
                    Text(
                        text = "No news selected.",
                        style = MaterialTheme.typography.body1,
                        modifier = Modifier.fillMaxWidth()
                    )
                }
            }
        }
    }
}

File: 2.0/watchOS/WatchSprings-2.0/WatchSprings Watch App/screens/NewsDetailScreen.swift
// WatchSprings 2.0 © 2025 Mark S. Morris. All rights reserved.
//
// https://www.watchsprings.com
//
// WatchSprings is Fan Content pursuant to the Epic Games Fan Content Policy. Portions of the materials used are trademarks and/or copyrighted works of Epic Games, Inc.

import SwiftUI
import Kingfisher

struct NewsDetailScreen: View
{
    @State var viewModel: NewsViewModel

    var body: some View
    {
        if let news = viewModel.selectedNews
        {
            ChildScreen(title: news.title ?? "News Detail", isLoading: false)
            {
                VStack(spacing: 16)
                {
                    if let url = URL(string: news.image ?? "News Detail")
                    {
                        KFImage(url)
                            .placeholder {
                                LoadingImage()
                            }
                            .resizable()
                            .aspectRatio(contentMode: .fit)
                    }
                    
                    Text(news.body ?? "News Detail")
                        .font(.body)
                        .padding()
                }
            }
            .padding()
        }
    }
}
